File: 1128-remove-all-adjacent-duplicates-in-string/1128-remove-all-adjacent-duplicates-in-string.java
class Solution {
    public String removeDuplicates(String s) {
        Stack<Character> stk = new Stack();
        for(char ch: s.toCharArray()){
            if(!stk.isEmpty() && stk.peek() == ch){
                stk.pop();
                continue;
            }else{
                stk.push(ch);
            }
        }
        if(!stk.isEmpty()){
            StringBuilder sb = new StringBuilder();
            while(!stk.isEmpty()){
                sb.append(stk.pop());
            }
            return sb.reverse().toString(); 
        } 
        return "";
    }
}

File: 1128-remove-all-adjacent-duplicates-in-string/1128-remove-all-adjacent-duplicates-in-string.rs
impl Solution {
    pub fn remove_duplicates(s: String) -> String {
        let mut array: Vec<char> = s.chars().collect();
        let mut end : isize = -1;
        for ch in s.chars(){
            if end >= 0 && array[end as usize] == ch{
                end -= 1;
            }else{
                end += 1;
                array[end as usize] = ch
            }
        }
        if end == -1{
            return "".to_string();
        }
        array[..=end as usize].iter().collect()
    }
}